src/test/crux-wallet-client.ts

```diff
@@ -0,0 +1,152 @@
+    InMemoryCruxUserRepository
+} from "./test-utils";
+
+chai.use(chaiAsPromised);
+chai.should();
+const expect = require('chai').expect;
+
+const testCruxDomain = getValidCruxDomain();
+const testCruxUser = getValidCruxUser();
+const testCruxUser2 = getValidCruxUser2();
+const testPvtKey = '6bd397dc89272e71165a0e7d197b280c7a88ed5b1e44e1928c25455506f1968f';  // 1HtFkbXFWHFW5Kd4GLfiRqkffS5KLZ91eJ
+const testPvtKey2 = '12381ab829318742938647283cd462738462873642ef34abefcd123501827193'; // 1JoZwbjMnTmcpAyjjtRBfuqXAb2xiqZRjx
+
+describe('CruxWalletClient Tests', function() {
+    beforeEach(function() {
+        this.inmemUserRepo = new InMemoryCruxUserRepository();
+        this.inmemDomainRepo = new InMemoryCruxDomainRepository();
+        addUserToRepo(testCruxUser, this.inmemUserRepo, new BasicKeyManager(testPvtKey));
+        addUserToRepo(testCruxUser2, this.inmemUserRepo, new BasicKeyManager(testPvtKey2));
+        addDomainToRepo(testCruxDomain, this.inmemDomainRepo);
+        this.stubGetCruxDomainRepository = sinon.stub(cwc, 'getCruxDomainRepository').callsFake(() => this.inmemDomainRepo as any);
+        this.stubGetCruxUserRepository = sinon.stub(cwc, 'getCruxUserRepository').callsFake(() => this.inmemUserRepo as any);
+    });
+    afterEach(function() {
+        this.stubGetCruxUserRepository.restore();
+        this.stubGetCruxDomainRepository.restore();
+    });
+    it('Nonexistent wallet name raises error', async function() {
+        let cc = new CruxWalletClient({
+            walletClientName: 'nonexistent'
+        });
+        const promise = cc.resolveCurrencyAddressForCruxID(testCruxUser.cruxID.toString(), 'bitcoin');
+        return expect(promise).to.be.eventually.rejected.with.property('errorCode', PackageErrorCode.InvalidWalletClientName);
+    });
+    describe('Resolving a Users ID', function() {
+        beforeEach(function() {
+            this.cc = new CruxWalletClient({
+                walletClientName: 'somewallet'
+            });
+        });
+        it('Happy case - valid users address', async function() {
+            const address = await this.cc.resolveCurrencyAddressForCruxID(testCruxUser.cruxID.toString(), 'bitcoin');
+            await expect(address).to.have.property('addressHash').equals('foobtcaddress');
+        });
+
+        it('Invalid ID', async function() {
+            const promise = this.cc.resolveCurrencyAddressForCruxID('lolwamax', 'bitcoin');
+            await expect(promise).to.be.eventually.rejected.with.property('errorCode', PackageErrorCode.CruxIdInvalidStructure);
+        });
+        it('Wallet doesnt have asset id mapped', async function() {
+            const promise = this.cc.resolveCurrencyAddressForCruxID(testCruxUser.cruxID.toString(), 'foo');
+            await expect(promise).to.be.eventually.rejected.with.property('errorCode', PackageErrorCode.AssetIDNotAvailable);
+        });
+        it('User doesnt have a currency address', async function() {
```

Comment:
is this behavior the same with the old version of SDK? does it not return None if address is not available with cruxUser.

Reply:
no, the behaviour is same - https://github.com/cruxprotocol/js-sdk/blob/master/src/index.ts#L229